wire/msgtx_test.go

```diff
@@ -779,6 +779,91 @@ func TestTxWitnessSize(t *testing.T) {
 	}
 }
 
+// TestTxWitnessLimits performs negative tests to ensure decoding
+// serialized transactions with witness data triggers proper error paths.
+func TestTxWitnessLimits(t *testing.T) {
+	// Use protocol version 70001 and transaction version 1 specifically
+	// here instead of the latest values because the test data is using
+	// bytes encoded with those versions.
+	pver := uint32(70001)
+	txVer := uint32(1)
+
+	var createTxnWithManyWitnessItems = func() []byte {
```

Comment:
Doesn't look to use many of the variables closed over, so I think we can just make this a regular function and then call it below.